pennylane/gradients/pulse_gradient.py

```diff
@@ -461,10 +532,51 @@ def ansatz(params):
     sampler_seed = sampler_seed or np.random.randint(18421)
     key = jax.random.PRNGKey(sampler_seed)
 
-    return _expval_stoch_pulse_grad(tape, argnum, num_split_times, key, shots)
+    return _expval_stoch_pulse_grad(tape, argnum, num_split_times, key, shots, use_broadcasting)
 
 
-def _expval_stoch_pulse_grad(tape, argnum, num_split_times, key, shots):
+def _generate_tapes_and_cjacs(tape, idx, key, num_split_times, use_broadcasting):
+    """Generate the tapes and compute the classical Jacobians for one given
+    generating Hamiltonian term of one pulse.
+    """
+    op, op_idx, term_idx = tape.get_operation(idx, return_op_index=True)
+    if not isinstance(op, ParametrizedEvolution):
+        raise ValueError(
+            "stoch_pulse_grad does not support differentiating parameters of "
+            "other operations than pulses."
+        )
+    if len(op.t) != 2:
+        raise ValueError(
+            "stoch_pulse_grad does not support differentiating ParametrizedEvolutions with "
+            f"intermediate time steps. Got t={op.t}"
+        )
+
+    coeff, ham = op.H.coeffs_parametrized[term_idx], op.H.ops_parametrized[term_idx]
+    if not qml.pauli.is_pauli_word(ham):
+        raise ValueError(
+            "stoch_pulse_grad currently only supports Pauli words as parametrized "
+            f"terms in ParametrizedHamiltonian. Got {ham}"
+        )
+    word = qml.pauli.pauli_word_to_string(ham)
```

Comment:
Why call the operator `ham` when it explicitly is not a Hamiltonian? Is a bit confusing for the reader. Maybe `obs`?